### ISSUE TITLE
Deleting the only control node in DO cluster , results in cluster getting stuck in "Updating" state forever.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#15810]
**Rancher versions: Build from master
**Steps to Reproduce:**

Create a DO cluster with following configuration:
node1 - Control plane
node2 - etcd 
node3 - Worker

Wait for cluster to become active.
Once the cluster is active , Delete node1.
node1 gets deleted successfully and new node gets added.
But the cluster gets stuck in "Updating" state forever with following error message:

```
	
This cluster is currently Updating; areas that interact directly with it will not be available until the API is ready.
Failed to delete controlplane node [167.99.55.199] from cluster: Get https://159.65.182.33:6443/api/v1/nodes?timeout=30s: dial tcp 159.65.182.33:6443: connect: connection refused
```

Complains about not being able to delete the controlplane node which has been already deleted.

**Comment by galal-hussein** [comment]
The problem was that rke can't remove the controlplane node from k8s if its the last controlplane node because it will not be able to reach kube-api, the fix handles this exact situation.

**Comment by sangeethah** [comment]
Tested with rancher-server version - v2.1.0-rc7

Still able to reproduce this issue with cluster being stuck in "Updating" state but the error message is different.

Had a DO cluster with following configuration:
1 control node
1 etcd node
2 worker nodes


Once cluster gets to "Active" state , delete the control node.
Control node gets deleted successfully and new node gets added which gets stuck in "registering" state with "Waiting to register with Kubernetes".
Cluster gets stuck in "Updating" state forever with following error message:

Note - santestc1 is the control plane node.
```
This cluster is currently Updating; areas that interact directly with it will not be available until the API is ready.

Failed to get kubernetes node [santestc1]: "santestc1" not found
```

**Comment by cjellick** [comment]
Is `santestc1` the old node or the new one?

**Comment by sangeethah** [comment]
@cjellick santestc1 is the name of the host in the nodepool. The old (deleted host) and the new one (the gets added automatically) gets the same name.

**Comment by cjellick** [comment]
Moving out of v2.1, can't make the cut off

**Comment by sangeethah** [comment]
This issue is not seen when testing with rancher server version 2.0.8 .

**Comment by sangeethah** [comment]
Tested with latest build from master.
Create a DO cluster with following configuration:
node1 - Control plane
node2 - etcd
node3 & node4 - Worker

Wait for cluster to become active.
Once the cluster is active , Delete node1.
node1 gets deleted successfully and new node gets added successfully and cluster gets back to "active" state.

**Comment by HangingClowns** [comment]
have this issue in #19916 or something similar

**Comment by tryhardest** [comment]
@sangeethah just curious why this was closed is #19916 not the same issue? I see the merge but still currently appears to exist, no?
